LevelEditor/SliderWidget.py

```diff
@@ -0,0 +1,182 @@
+from PySide.QtCore import *
+from PySide.QtGui import *
+from collections import OrderedDict
+
+sliders = None
+
+
+class AdjustValues(QWidget):
+    valuesChanged = Signal(list)
+
+    def __init__(self, val_list, min_val, max_val):
+        super(AdjustValues, self).__init__()
+        self.val_list = val_list
+        self.min_val = min_val
+        self.max_val = max_val
+        self.slider_count = len(val_list)
+        self.slides = OrderedDict()
+
+        # Insert Sliders
+        self.wrap = QVBoxLayout()
+
+        for i in range(len(val_list)):
+            row = QHBoxLayout()
+            lab = QLabel(str(val_list[i]))
+            slide = QSlider(Qt.Horizontal)
+
+            # remove button
+            rem_button = QPushButton()
+            rem_button.setText("X")
+            rem_button.clicked.connect(self.remove_level(row))
+
+            # setting slider attributes
+            slide.setRange(min_val, max_val)
+            slide.setTickPosition(QSlider.TicksAbove)
+            slide.sliderMoved.connect(self.change_label(lab, slide))
+            slide.sliderMoved.connect(self.adjust_slides(slide))
+            slide.sliderMoved.connect(self.send_values)
+            slide.setValue(val_list[i])
+
+            # label = l, slider = s, position = p
+
+            d = {"l": lab, "s": slide, "p": val_list[i]}
+
+            self.slides[slide] = d
+            row.addWidget(rem_button)
+            row.addWidget(lab)
+            row.addWidget(slide)
+            self.wrap.addLayout(row)
+
+        # Add Line Button
+        self.add_line = QPushButton()
+        self.add_line.setText("+ Add Level")
+        self.add_line.clicked.connect(self.insert_line)
+        self.wrap.addWidget(self.add_line)
+
+        # add layout
+        self.setLayout(self.wrap)
+
+    def moved_forward(self, cur_slide):
```

Comment:
You should be able to refactor both `moved_forward` and `moved_backward` into one function; something like:

```python
def moved_slider(self, slider_index):
```

and then check if the sliders with a lower index are >= slider_index's value, and the sliders with a higher index are <= slider_index's value.